### PR TITLE
FE: Generate grpc server and create python app

### PR DESCRIPTION
**Pull request by smacker**
Part of #51: 

- Generation scripts: server in python src/main/python/generated.py
- Python CLI app for FE at `src/main/python/server.py` configures the port

I put generation script in `src/main/resources/generate_from_proto.sh` for now but we might need a better place (root Makefile?).

### CODE REVIEW

**Comment by bzz** [comment]
Looks nice, 2 cosmetic things:

 1. do you think it could be worth to break down this commit to two: generated files, and everything else? This way the history/blame in Git will be very clear.

 2. `src/main/python` is going to be the place for multiple python apps, i.e CLI for community detection will also live there. Knowing that, do you it may be worth using something different from `main.py` for feature extractors?

File: src/main/python/generate.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (bzz) [inline src/main/python/generate.sh]: I suggest to keep this .sh script in `src/main/resources`, as it's follows the standard project layout, and most probably this script will also include generating gRPC/Protobuf for other languages.

**Comment by bzz** [comment]
One more thing - it's totally fine at this point not to have even a basic tests, but I think we should update a developer documentation https://github.com/src-d/gemini#dev  

Adding one more section about gRPC/Protobuf generation, pointing to `src/main/resourcese/generate.sh` and briefly mentioning what is the input and what is the output of this process and when it should be run.

File: src/main/python/generate.sh (hunk 1 above)
Comment (bzz) [reply src/main/python/generate.sh]: And it's also good idea to add a small comment here, following https://google.github.io/styleguide/shell.xml#File_Header

**Comment by smacker** [comment]
Thanks @bzz! Those aren't nitpicking at all!

About tests:
I couldn't figure out what I could test on this point (all methods are raising `NotImplementedError` exception for now). I think it's better to add tests later when there will be implementation.

**Comment by smacker** [comment]
I split code into 3 different commits:
- a script for code generation
- generated code
- app

Moved the script to resources directory.

Renamed main.py to server.py (currently we are going to have only one server, later we may move this file to separate folder inside python, but for now I would like to keep it as simple as possible)

**Comment by smacker** [comment]
I added doc in next PR: https://github.com/src-d/gemini/pull/63

**Comment by bzz** [comment]
Took liberty of updating PR description

**Comment by bzz** [comment]
@carlosms on 

> Instead of requirements.txt, put the requirements in the readme

this file is machine-readable and is part of standard tooling for dependency management on Python with `pip install -r requirements.txt`, so no need to duplicate this in human-readable format IMO.

> Add a stdout message when the server starts, something like 'started listening on port xx'.

👍 not sure if that blocks the merge, but even simplest logging for such things is a very nice engineering practice indeed, that IMO is worth adding at from the beginning of the project

**Comment by smacker** [comment]
added logging

**Pull request merged**